times_excel_reader.py

```diff
@@ -1651,86 +1695,222 @@ def dump_tables(tables: List, filename: str) -> List:
     return tables
 
 
-def convert_xl_to_times(
-    dir: str, input_files: List[str], mappings: List[TimesXlMap]
-) -> Dict[str, DataFrame]:
-    pickle_file = "raw_tables.pkl"
-    if os.path.isfile(pickle_file):
-        raw_tables = pickle.load(open(pickle_file, "rb"))
-        print(f"WARNING: Using pickled data not xlsx")
-    else:
-        raw_tables = []
-        filenames = [os.path.join(dir, filename) for filename in input_files]
+def expand_rows_parallel(tables: List[EmbeddedXlTable]) -> List[EmbeddedXlTable]:
+    with ProcessPoolExecutor() as executor:
+        return list(executor.map(expand_rows, tables))
 
-        use_pool = True
-        if use_pool:
-            with ProcessPoolExecutor() as executor:
-                for result in executor.map(extract_tables, filenames):
-                    raw_tables.extend(result)
-        else:
-            for f in filenames:
-                result = extract_tables(f)
-                raw_tables.extend(result)
-        pickle.dump(raw_tables, open(pickle_file, "wb"))
 
-    print(
-        f"Extracted {len(raw_tables)} tables,"
-        f" {sum(table.dataframe.shape[0] for table in raw_tables)} rows"
-    )
+# ============================================================================
+# ===============           HELPER FUNCTIONS                 =================
+# ============================================================================
 
-    transforms = [
-        lambda tables: dump_tables(tables, "raw_tables.txt"),
-        normalize_tags_columns_attrs,
-        remove_fill_tables,
-        lambda tables: [remove_comment_rows(t) for t in tables],
-        lambda tables: [remove_comment_cols(t) for t in tables],
-        remove_tables_with_formulas,  # slow
-        process_transform_insert,
-        process_flexible_import_tables,  # slow
-        process_user_constraint_tables,
-        process_commodity_emissions,
-        process_commodities,
-        process_processes,
-        process_transform_availability,
-        fill_in_missing_values,
-        process_time_slices,
-        expand_rows_parallel,  # slow
-        remove_invalid_values,
-        process_time_periods,
-        process_currencies,
-        apply_fixups,
-        extract_commodity_groups,
-        merge_tables,
-        process_datayears,
-        process_wildcards,
-        convert_to_string,
-        lambda tables: dump_tables(tables, "merged_tables.txt"),
-        lambda tables: produce_times_tables(tables, mappings),
+
+def apply_composite_tag(table: EmbeddedXlTable) -> EmbeddedXlTable:
+    """
+    If the table tag is composite (contains ':', e.g. ~FI_T: COM_PKRSV)
+    fill in the 'Atribute' field in the dataframe with the tag value
+    (e.g. COM_PKRSV ) and return a modified version of the table with a
+    simplified table tag (e.g. ~FI_T)
+    If the table tag is not composite, return the unmodified table.
+    TODO: don't really understand why we are doing this.
```

Comment:
Handles "Table level declarations", described on page 15 of https://iea-etsap.org/docs/Documentation_for_the_TIMES_Model-Part-IV.pdf